server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -148,7 +150,14 @@ private void doExecuteForked(
             // in the case we have one or more remote indices but no local we don't expand to all local indices and just do remote indices
             concreteIndices = Strings.EMPTY_ARRAY;
         } else {
-            concreteIndices = indexNameExpressionResolver.concreteIndexNames(clusterState, localIndices);
+            boolean ignoreUnavailable = request.indicesOptions().ignoreUnavailable();
+            concreteIndices = Arrays.stream(indexNameExpressionResolver.concreteIndexNames(clusterState, localIndices)).filter(index -> {
+                IndexMetadata metadata = clusterState.getMetadata().getIndices().get(index);
+                if (metadata != null && metadata.getState() == IndexMetadata.State.CLOSE) {
+                    if (ignoreUnavailable) return false;
+                    else throw new IndexClosedException(metadata.getIndex());
+                } else return true;
```

Comment:
I am a bit outdated on how indices resolution works these days, after recent refactoring. I was somehow not expecting that you need to explicitly handle closed indices, rather that provided indices and indices options, the method does what it needs to do, meaning not return closed indices. Can you double check that that option is no longer possible?

Reply:
Thank you, @javanna, for reviewing the pr. You are right. Further debugging the code, I found that for the incoming requests, the **allowClosedIndices** is set to **true**. Changing the default value at the FieldCapabilitiesRequest level, mitigates that issue.